### PR TITLE
Updated singular tests section to show they can now be documented

### LINKED ISSUE #6132: [Core] Singular data tests can now be documented
At a minimum, update body to include a link to the page on docs.getdbt.com requiring updates and what part(s) of the page you would like to see updated.
Originating from this issue: https://github.com/dbt-labs/dbt-core/issues/9005
~Original pull request: https://github.com/dbt-labs/dbt-core/pull/10744~
New pull request: https://github.com/dbt-labs/dbt-core/pull/10792

## Page(s) to update

- https://docs.getdbt.com/reference/resource-properties/description (add new "Data tests" tab)
- https://docs.getdbt.com/docs/build/data-tests (maybe 🤷)
- https://docs.getdbt.com/reference/project-configs/docs-paths (since it now includes [`test-paths`](https://docs.getdbt.com/reference/project-configs/test-paths))

## Code examples

`tests/my_singular_test.sql`

```sql
with my_cte as (
    select 1 as id, 'foo' as name
    union all
    select 2 as id, 'bar' as name
)
select * from my_cte
```

### Without docs blocks

`tests/_properties.yml`

```yaml
data_tests:
  - name: my_singular_test
    description: Some docs
    config:
      error_if: ">10"
    meta:
      some_key: some_val
```

### With docs blocks

Or with docs blocks:

`tests/example.md`

```md
{% docs my_singular_test_documentation %}
Some docs from a doc block
{% enddocs %}
```

`tests/_properties.yml`

```yaml
data_tests:
  - name: my_singular_test
    description: "{{ doc('my_singular_test_documentation') }}"
    config:
      error_if: ">10"
    meta:
      some_key: some_val
```

### List 'em and run 'em

List 'em:

```shell
$ dbt list --resource-type test -q 

my_project.my_singular_test
```

Run 'em:

```shell
$ dbt test -s my_project.my_singular_test

19:55:23  Running with dbt=1.9.0-a1
19:55:23  Registered adapter: postgres=1.9.0-a1
19:55:23  Found 1 test, 423 macros
19:55:23  
19:55:23  Concurrency: 5 threads (target='postgres')
19:55:23  
19:55:24  1 of 1 START test my_singular_test ............................................. [RUN]
19:55:24  1 of 1 FAIL 2 my_singular_test ................................................. [FAIL 2 in 0.10s]
19:55:24  
19:55:24  Finished running 1 test in 0 hours 0 minutes and 1.13 seconds (1.13s).
19:55:24  
19:55:24  Completed with 1 error, 0 partial successs, and 0 warnings:
19:55:24  
19:55:24  Failure in test my_singular_test (tests/my_singular_test.sql)
19:55:24    Got 2 results, configured to fail if != 0
19:55:24  
19:55:24    compiled code at target/compiled/my_project/tests/my_singular_test.sql
19:55:24  
19:55:24  Done. PASS=0 WARN=0 ERROR=1 SKIP=0 TOTAL=1
```

### PR DESCRIPTION
**Pull request by nataliefiann**
Resolves https://github.com/dbt-labs/docs.getdbt.com/issues/6132

## What are you changing in this pull request and why?

The `description` property is available for singular data tests beginning in dbt v1.9, and this PR highlights how singular tests can now be documented.

### 🎩 Previews
- [document singular data tests](https://docs-getdbt-com-git-nfiann-singulartests-dbt-labs.vercel.app/docs/build/data-tests#document-singular-tests)
- [`description`](https://docs-getdbt-com-git-nfiann-singulartests-dbt-labs.vercel.app/reference/resource-properties/description)
- [`docs-paths`](https://docs-getdbt-com-git-nfiann-singulartests-dbt-labs.vercel.app/reference/project-configs/docs-paths)
- [documentation placement](https://docs-getdbt-com-git-nfiann-singulartests-dbt-labs.vercel.app/docs/build/documentation#placement)

## Checklist
- [x] I have reviewed the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [x] The topic I'm writing about is for specific dbt version(s) and I have versioned it according to the [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and/or [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content) guidelines.
- [ ] Wait until https://github.com/dbt-labs/dbt-core/pull/10792 is merged
- [x] Add a note to the prerelease version [Migration Guide](https://github.com/dbt-labs/docs.getdbt.com/tree/current/website/docs/docs/dbt-versions/core-upgrade)

### CODE REVIEW

File: website/docs/docs/build/data-tests.md
Comment (mirnawong1) [inline website/docs/docs/build/data-tests.md]:
```suggestion
Previously, documenting a data test such as a singular data test would result in a parsing error. However, in [Versionless](/dbt-versions/upgrade-dbt-version-in-cloud#versionless), you can set descriptions in your YAML files to document singular data tests.
```

**Review by mirnawong1 — APPROVED** [review]
lgtm, thanks @nataliefiann ! note, i would recommend writing 'Resolves # issue_number' in the pr description. this will automatically close the related issue once the pr is merged. in this case, the issue is [#6132](https://github.com/dbt-labs/docs.getdbt.com/issues/6132)

File: website/docs/docs/build/data-tests.md
Comment (graciegoheen) [inline website/docs/docs/build/data-tests.md]: I don't think we need to callout previous behavior. Instead, a callout that this is available in versionless and will be available in dbt Core `v1.9` and then just a description of the feature.

File: website/docs/docs/build/data-tests.md
Comment (graciegoheen) [inline website/docs/docs/build/data-tests.md]: @aranke Can you confirm that docs blocks will work for documenting singular tests as well?

**Review by graciegoheen — CHANGES_REQUESTED** [review]
Just a heads up - we had to rollback this change. Kshitij is working on a fix, but we should hold off on merging this until that is complete. 

See -> https://github.com/dbt-labs/dbt-core/issues/9005

**Comment by nataliefiann** [comment]
Log: moving this blocked in Kanban until status update from engineering

File: website/docs/docs/build/data-tests.md
Comment (dbeatty10) [inline website/docs/docs/build/data-tests.md]:
```suggestion
You can set descriptions in your YAML files to document singular data tests:
```

File: website/docs/docs/build/data-tests.md
Comment (dbeatty10) [reply website/docs/docs/build/data-tests.md]: ## TLDR
We have two issues open for adding `description` to data tests:
- https://github.com/dbt-labs/dbt-core/issues/9005#issuecomment-1834047344
- https://github.com/dbt-labs/dbt-core/issues/2578

This PR _only_ allows the first and not the second.

## Details

There are two types of data tests:
1. Generic
1. Singular

### Descriptions for generic data tests

Generic data tests have one macro _definition_ and then zero to many _instances_ where they are used. So that's two different places we can imagine adding a description for generic data tests. My understanding is the first is supported by this PR but the second is **not**.

Since the _definition_ is just a macro, the definition of a custom generic data test can already have a [`description`](https://docs.getdbt.com/reference/resource-properties/description) like this:

```yaml
macros:
  - name: test_CUSTOM_GENERIC_TEST_NAME
    description: markdown_string
    arguments:
      - name: argument_name
        description: markdown_string
```

However, I don't believe the _instances_ of generic data tests can be documented until https://github.com/dbt-labs/dbt-core/issues/2578 is implemented.

### Descriptions for singular data tests

Added in https://github.com/dbt-labs/dbt-core/pull/10792 to resolve https://github.com/dbt-labs/dbt-core/issues/9005!

I will test out this example to make sure it works 👍

Comment (dbeatty10) [reply website/docs/docs/build/data-tests.md]: ✅ Works

File: website/docs/docs/build/data-tests.md
Comment (dbeatty10) [inline website/docs/docs/build/data-tests.md]: While any of the [`docs-paths`](https://docs.getdbt.com/reference/project-configs/docs-paths) will work, the `tests` directory makes sense here because that is the default for [`test-paths`](https://docs.getdbt.com/reference/project-configs/test-paths). Furthermore, using `singular` not because it is required (it's not!), but because it is a very readable and logical complement to the `generic` subdirectory that contains definitions for custom generic data tests.

```suggestion
<File name='tests/singular/data_tests.yml'>
```

File: website/docs/docs/build/documentation.md
Comment (mirnawong1) [inline website/docs/docs/build/documentation.md]:
```suggestion
Docs blocks should be placed in files with a `.md` file extension. By default, dbt will search in all resource paths for docs blocks (for example, the combined list of [model-paths](/reference/project-configs/model-paths), [seed-paths](/reference/project-configs/seed-paths), [analysis-paths](/reference/project-configs/analysis-paths), [test-paths](/reference/project-configs/test-paths), [macro-paths](/reference/project-configs/macro-paths) and [snapshot-paths](/reference/project-configs/snapshot-paths)) &mdash; you can adjust this behavior using the [docs-paths](/reference/project-configs/docs-paths) config.
```

File: website/docs/docs/build/documentation.md
Comment (mirnawong1) [inline website/docs/docs/build/documentation.md]:
```suggestion
Docs blocks should be placed in files with a `.md` file extension. By default, dbt will search in all resource paths for docs blocks (for example, the combined list of [model-paths](/reference/project-configs/model-paths), [seed-paths](/reference/project-configs/seed-paths), [analysis-paths](/reference/project-configs/analysis-paths), [macro-paths](/reference/project-configs/macro-paths) and [snapshot-paths](/reference/project-configs/snapshot-paths)) &mdash; you can adjust this behavior using the [docs-paths](/reference/project-configs/docs-paths) config.
```

File: website/docs/docs/build/documentation.md
Comment (mirnawong1) [inline website/docs/docs/build/documentation.md]:
```suggestion
Docs blocks should be placed in files with a `.md` file extension. By default, dbt will search in all resource paths for docs blocks (for example, the combined list of [model-paths](/reference/project-configs/model-paths), [seed-paths](/reference/project-configs/seed-paths), [analysis-paths](/reference/project-configs/analysis-paths), [test-paths](/reference/project-configs/test-paths), [macro-paths](/reference/project-configs/macro-paths), and [snapshot-paths](/reference/project-configs/snapshot-paths)) &mdash; you can adjust this behavior using the [docs-paths](/reference/project-configs/docs-paths) config.
```

File: website/docs/docs/build/documentation.md
Comment (mirnawong1) [inline website/docs/docs/build/documentation.md]:
```suggestion
Docs blocks should be placed in files with a `.md` file extension. By default, dbt will search in all resource paths for docs blocks (for example, the combined list of [model-paths](/reference/project-configs/model-paths), [seed-paths](/reference/project-configs/seed-paths), [analysis-paths](/reference/project-configs/analysis-paths), [macro-paths](/reference/project-configs/macro-paths), and [snapshot-paths](/reference/project-configs/snapshot-paths)) &mdash; you can adjust this behavior using the [docs-paths](/reference/project-configs/docs-paths) config.
```

File: website/docs/reference/resource-properties/description.md
Comment (mirnawong1) [inline website/docs/reference/resource-properties/description.md]:
```suggestion
The `description` property is available for [singular data tests](/build/data-tests#document-singular-data-tests) beginning in dbt v1.9.
```

File: website/docs/docs/build/data-tests.md
Comment (mirnawong1) [inline website/docs/docs/build/data-tests.md]:
```suggestion
The `description` property is available for [singular data tests](/build/data-tests#document-singular-data-tests) beginning in dbt v1.9.
```

**Review by mirnawong1 — APPROVED** [review]
lgtm and i think it's good to merge since [PR 10792](https://github.com/dbt-labs/dbt-core/pull/10792) has been merged @dbeatty10

**Comment by nataliefiann** [comment]
Closed off as Core project has been paused

**Comment by dbeatty10** [comment]
Moved to https://github.com/dbt-labs/docs.getdbt.com/pull/6191

**Pull request closed without merging**
